packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -184,6 +184,14 @@ describe('<DataGrid /> - Filter', () => {
         expect(getColumnValues()).to.deep.equal(expected);
       });
     });
+
+    ['contains', 'startsWith', 'endsWith'].forEach((operatorValue) => {
+      it('should escape RegExp characters if applied as filter values', () => {
+        const regExpToEscape = '[-[]{}()*+?.,\\^$|#s]';
+        render(<TestCase value={regExpToEscape} operatorValue={operatorValue} />);
+        expect(getColumnValues()).to.deep.equal([]);
```

Comment:
I don't think that it's enough, we should still be able to find the result.

Reply:
If the assertion is checking `[]`, it is ignoring the input value and returning zero rows if it is a regex. So what do we want to support? Should we escape regex and show empty or able to filter without any crash even if it is a RegExp? Like support * as all rows?
Also I checked with Commodity `contains` filter one case wherein `soy*` worked before as `starts with soy` and now it simply returns zero rows for `soy*` in Xgrid demo.  I will have to check that.

Reply:
IMO soy* should just search for soy*? So we don't support regex search
Otherwise if a value is *, it is not filterable

Reply:
> IMO soy* should just search for soy*

This works with the current fix. 
Example: If a row contains `soy*bean` value string, filter operation `contains` `soy*` will return this row. What happens is the filter regex is now `soy\*bean` making * as literal instead of regex special. 
 I could add a test case for this.

Reply:
Yes it's the correct behaviour. And yes please add a test so it's a bit more robust.

Reply:
I added one.